loader/loader.go

```diff
@@ -340,7 +341,7 @@ func (l *Loader) parseGunkPackage(pkg *GunkPackage) {
 		if pkg.ProtoName == "" {
 			pkg.ProtoName = name
 		} else if name != "" {
```

Comment:
I had a much simpler solution in mind, which perhaps I should have clarified on the ticket.

We can simply skip the package name mismatch check if `l.Types == false`. We don't need to typecheck, so we don't need to follow the restriction.

In technical terms a package should still contain only the files for that package. But in practice, we don't care all that much, as the only commands that load packages are format (syntax-only and per-file) and generate (with `Types == true`). If we need the current behavior in the future (correct packages with sanity checks, but no type-checking), we can always add more knobs to Loader.

Reply:
This also means that you can remove much of the added code in format.go.

Reply:
done, thanks!